corehq/apps/domain/forms.py

```diff
@@ -1157,3 +1167,430 @@ def process_submission(self, domain=None):
+            self.fields['end_date'].initial = self.current_subscription.date_end
+            self.helper.layout = crispy.Layout(
+                TextField('software_plan_edition', plan_edition),
+                crispy.Hidden('software_plan_edition', plan_edition),
+                crispy.Field('fogbugz_client_name'),
+                crispy.Field('emails', css_class='input-xxlarge'),
+                TextField('start_date', self.current_subscription.date_start),
+                crispy.Hidden('start_date', self.current_subscription.date_start),
+                crispy.Field('end_date', css_class='date-picker'),
+                crispy.Hidden('sms_credits', 0),
+                crispy.Hidden('user_credits', 0),
+                crispy.HTML(_(
+                    '<div class="alert">'
+                    '<p><strong>Are you sure you want to extend the subscription?</strong></p>'
+                    '<p>If this project is becoming a self-service project and only paying for '
+                    'hosting fees, please have them self-subscribe through the subscription page.  '
+                    'Please use this page only to extend the existing services contract.</p>'
+                    '</div>'
+                )),
+                self.form_actions
+            )
+
+    def process_subscription_management(self):
+        new_plan_version = DefaultProductPlan.get_default_plan_by_domain(
+            self.domain, edition=self.cleaned_data['software_plan_edition'],
+        )
+        revert_current_subscription_end_date = None
+        if self.current_subscription and self.cleaned_data['start_date'] < self.current_subscription.date_end:
+            revert_current_subscription_end_date = self.current_subscription.date_end
+            self.current_subscription.date_end = self.cleaned_data['start_date']
+            self.current_subscription.save()
+        try:
+            if not self.current_subscription or self.cleaned_data['start_date'] > datetime.date.today():
+                new_subscription = Subscription.new_domain_subscription(
+                    self.next_account,
+                    self.domain,
+                    new_plan_version,
+                    date_start=self.cleaned_data['start_date'],
+                    date_end=self.cleaned_data['end_date'],
+                    web_user=self.web_user,
+                )
+            else:
+                new_subscription = self.current_subscription.change_plan(
+                    new_plan_version,
+                    date_end=self.cleaned_data['end_date'],
+                    web_user=self.web_user,
+                )
+            if new_subscription.date_start <= datetime.date.today() and datetime.date.today() < new_subscription.date_end:
+                new_subscription.is_active = True
+            new_subscription.do_not_invoice = False
+            new_subscription.auto_generate_credits = True
+            new_subscription.save()
+        except:
```

Comment:
Not sure if it makes sense here, but would be nice to either log an error so we know what went wrong or display a message to the user